haskell/defs.bzl

```diff
@@ -309,6 +315,10 @@ def haskell_binary(
       srcs: Haskell source files. File names must match module names, see above.
       extra_srcs: Extra (non-Haskell) source files that will be needed at compile time (e.g. by Template Haskell).
       deps: List of other Haskell libraries to be linked to this target.
+      narrowed_deps: Like deps, but only for dependencies using the modules
+          attribute. These dependencies are only used if this library uses
+          the modules attribute and the haskell_module rules depend on modules
+          provided by these dependencies.
       modules: List of extra haskell_module() dependencies to be linked into this binary.
```

Comment:
`haskell_module` is still marked as experimental. Let's also mark `narrowed_deps` and `modules` as experimental while that is the case.

Reply:
I added notes to both attributes saying that they are experimental.

Reply:
Could you also add such notes to the other instances at `haskell_library` and `haskell_test`?

Reply:
Done. I forgot that this piece of documentation was duplicated.